### PR TITLE
Use adjust android sdk as submodule

### PR DESCRIPTION
**Pull request by mbirghan**

### CODE REVIEW

File: scripts/android_package.sh [hunk 1]
```diff
@@ -128,6 +128,13 @@ print Y "Configuring the android build"
 if [[ "$ADJUST_SDK_TOKEN" ]]; then
   print Y "Use adjust config"
   ADJUST="CONFIG+=adjust"
+
+  command -v find >/dev/null 2>&1 || die "find must be installed to work with the Adjust SDK"
+  command -v sed >/dev/null 2>&1 || die "sed must be installed to work with the Adjust SDK"
+  rm -rf "android/src/com/adjust"
+  cp -a "3rdparty/adjust-android-sdk/Adjust/sdk-core/src/main/java/com/." "android/src/com/"
+  find "android/src/com/adjust" -name *.java -exec sed -i "" "s/javax.net.ssl.HttpsURLConnection/java.net.HttpURLConnection/g" {} \;
+  find "android/src/com/adjust" -name *.java -exec sed -i "" "s/ttps/ttp/g" {} \;
```
Comment (bakulf) [inline scripts/android_package.sh:137]: No, this is not good :/ We need to replace https more carefully. It seems that we just need to apply this patch:

```
diff --git a/Adjust/sdk-core/src/main/java/com/adjust/sdk/AdjustLinkResolution.java b/Adjust/sdk-core/src/main/java/com/adjust/sdk/AdjustLinkResolution.java
index 668fa20..2317e3c 100644
--- a/Adjust/sdk-core/src/main/java/com/adjust/sdk/AdjustLinkResolution.java
+++ b/Adjust/sdk-core/src/main/java/com/adjust/sdk/AdjustLinkResolution.java
@@ -105,11 +105,11 @@ public final class AdjustLinkResolution {
                                           final int recursionNumber,
                                           final AdjustLinkResolutionCallback adjustLinkResolutionCallback)
     {
-        final URL httpsUrl = convertToHttps(urlToRequest);
+        final URL httpUrl = convertToHttp(urlToRequest);
         URL resolvedURL = null;
         HttpURLConnection ucon = null;
         try {
-            ucon = (HttpURLConnection) httpsUrl.openConnection();
+            ucon = (HttpURLConnection) httpUrl.openConnection();
             ucon.setInstanceFollowRedirects(false);
 
             ucon.connect();
@@ -126,7 +126,7 @@ public final class AdjustLinkResolution {
             }
 
             resolveLink(resolvedURL,
-                    httpsUrl,
+                    httpUrl,
                     recursionNumber + 1,
                     adjustLinkResolutionCallback);
         }
@@ -154,7 +154,7 @@ public final class AdjustLinkResolution {
         return false;
     }
 
-    private static URL convertToHttps(final URL urlToConvert) {
+    private static URL convertToHttp(final URL urlToConvert) {
         if (urlToConvert == null) {
             return urlToConvert;
         }
@@ -171,7 +171,7 @@ public final class AdjustLinkResolution {
 
         URL convertedUrl = urlToConvert;
         try {
-            convertedUrl = new URL("https:" + stringUrlToConvert.substring(5));
+            convertedUrl = new URL("http:" + stringUrlToConvert.substring(5));
         } catch (final MalformedURLException ignored) { }
 
         return convertedUrl;
diff --git a/Adjust/sdk-core/src/main/java/com/adjust/sdk/Constants.java b/Adjust/sdk-core/src/main/java/com/adjust/sdk/Constants.java
index c8f1a0e..2f07c2e 100644
--- a/Adjust/sdk-core/src/main/java/com/adjust/sdk/Constants.java
+++ b/Adjust/sdk-core/src/main/java/com/adjust/sdk/Constants.java
@@ -27,7 +27,7 @@ public interface Constants {
     String GDPR_URL = "https://gdpr.adjust.com";
     String SUBSCRIPTION_URL = "https://subscription.adjust.com";
 
-    String SCHEME = "https";
+    String SCHEME = "http";
     String AUTHORITY = "app.adjust.com";
     String CLIENT_SDK = "android4.28.5";
     String LOGTAG = "Adjust";
``

**Review by bakulf — CHANGES_REQUESTED** [review]

File: scripts/android_package.sh (hunk 1 above)
Comment (mbirghan) [reply scripts/android_package.sh:137]: Did you try to build with that patch?

**Comment by mbirghan** [comment]
I had to create a larger patch because we have to replace HTTPS at multiple places including the imports.

File: scripts/android_package.sh [hunk 2]
```diff
@@ -128,6 +128,10 @@ print Y "Configuring the android build"
 if [[ "$ADJUST_SDK_TOKEN" ]]; then
   print Y "Use adjust config"
   ADJUST="CONFIG+=adjust"
+
+  rm -rf "android/src/com/adjust"
```
Comment (bakulf) [inline scripts/android_package.sh:132]: `|| die "failed to remove the adjust folder"`

File: scripts/android_package.sh [hunk 3]
```diff
@@ -128,6 +128,10 @@ print Y "Configuring the android build"
...
+  cp -a "3rdparty/adjust-android-sdk/Adjust/sdk-core/src/main/java/com/." "android/src/com/"
```
Comment (bakulf) [inline scripts/android_package.sh:133]: `|| die "Failed to copy the adjust codebase"` or something similar

File: scripts/android_package.sh [hunk 4]
```diff
@@ -128,6 +128,10 @@ print Y "Configuring the android build"
...
+  git apply --directory="android/src/" "3rdparty/adjust_https_to_http.diff"
```
Comment (bakulf) [inline scripts/android_package.sh:134]: `|| die "Failed to apply the patch"` or something better.

File: 3rdparty/adjust_https_to_http.diff [hunk 5]
```diff
@@ -0,0 +1,339 @@
...
+diff --git a/com/adjust/sdk/AdjustLinkResolution.java b/com/adjust/sdk/AdjustLinkResolution.java
+index 668fa208..b22ae305 100644
+--- a/com/adjust/sdk/AdjustLinkResolution.java
++++ b/com/adjust/sdk/AdjustLinkResolution.java
+@@ -13,7 +13,7 @@ public final class AdjustLinkResolution {
+         void resolvedLinkCallback(Uri resolvedLink);
+     }
+ 
+-    // https://www.cs.umd.edu/~pugh/java/memoryModel/DoubleCheckedLocking.html
++    // http://www.cs.umd.edu/~pugh/java/memoryModel/DoubleCheckedLocking.html
```
Comment (bakulf) [inline 3rdparty/adjust_https_to_http.diff:81]: this no :)

File: 3rdparty/adjust_https_to_http.diff [hunk 6]
```diff
@@ -0,0 +1,339 @@
...
+--- a/com/adjust/sdk/Constants.java
++++ b/com/adjust/sdk/Constants.java
+@@ -23,11 +23,11 @@ public interface Constants {
+     int SOCKET_TIMEOUT = Constants.ONE_MINUTE;
+     int MAX_WAIT_INTERVAL = Constants.ONE_MINUTE;
+ 
+-    String BASE_URL = "https://app.adjust.com";
+-    String GDPR_URL = "https://gdpr.adjust.com";
+-    String SUBSCRIPTION_URL = "https://subscription.adjust.com";
++    String BASE_URL = "http://app.adjust.com";
```
Comment (bakulf) [inline 3rdparty/adjust_https_to_http.diff:137]: let's keep these 3

File: 3rdparty/adjust_https_to_http.diff [hunk 7]
```diff
@@ -0,0 +1,339 @@
...
+             logger.error("Response message: %s", responseMessage);
+diff --git a/com/adjust/sdk/network/UrlStrategy.java b/com/adjust/sdk/network/UrlStrategy.java
+index 19f6be46..b1def020 100644
+--- a/com/adjust/sdk/network/UrlStrategy.java
++++ b/com/adjust/sdk/network/UrlStrategy.java
+@@ -14,25 +14,25 @@ import static com.adjust.sdk.AdjustConfig.URL_STRATEGY_INDIA;
+ import static com.adjust.sdk.AdjustConfig.DATA_RESIDENCY_EU;
+ 
+ public class UrlStrategy {
+-    private static final String BASE_URL_INDIA = "https://app.adjust.net.in";
```
Comment (bakulf) [inline 3rdparty/adjust_https_to_http.diff:250]: these are OK!

File: 3rdparty/adjust_https_to_http.diff [hunk 8]
```diff
@@ -0,0 +1,339 @@
...
+-    private static final String GDPR_URL_INDIA = "https://gdpr.adjust.net.in";
+-    private static final String SUBSCRIPTION_URL_INDIA = "https://subscription.adjust.net.in";
++    private static final String BASE_URL_INDIA = "http://app.adjust.net.in";
++    private static final String GDPR_URL_INDIA = "http://gdpr.adjust.net.in";
++    private static final String SUBSCRIPTION_URL_INDIA = "http://subscription.adjust.net.in";
+ 
+-    private static final String BASE_URL_CHINA = "https://app.adjust.world";
```
Comment (bakulf) [inline 3rdparty/adjust_https_to_http.diff:257]: here too

**Review by bakulf — CHANGES_REQUESTED** [review]

**Review by bakulf — APPROVED** [review]

**Pull request merged**